src/ripple/basics/Slice.h

```diff
@@ -34,34 +37,64 @@
 
 namespace ripple {
 
-/** An immutable linear range of bytes.
+enum class SliceImplMutability : bool { no, yes };
+
+/** A linear range of bytes.
 
     A fully constructed Slice is guaranteed to be in a valid state.
     A Slice is lightweight and copyable, it retains no ownership
     of the underlying memory.
 */
-class Slice
+template<SliceImplMutability M>
```

Comment:
This `SliceImplMutability` business seems like too much effort and a bit too convoluted. Why not have something like this:

    template <typename T,
        typename = std::enable_if_t<std::is_same<std::remove_cv_t<T>, std::uint8_t>::value>
    >
    class SliceImpl
    {
        T* data_ = nullptr;
    };

    /** An immutable linear range of bytes. */
    using Slice = SliceImpl<std::uint8_t const>;
    /** A mutable linear range of bytes. */
    using MutableSlice = SliceImpl<std::uint8_t>;

Reply:
Addressed in bcaaf38a3e0e9516fd5ac28d73a7721de73188c0

BTW, I changed the `enable_if` into a `static_assert`. I'd use `enable_if` for an overload-set. For an assumption that I want to error out I'd rather use `static_assert`.